addons/html_editor/static/src/main/link/link_popover.js

```diff
@@ -158,6 +172,20 @@ export class LinkPopover extends Component {
         this.onClickApply();
     }
 
+    onClickForceEditMode(ev) {
+        if (this.props.linkElement.href) {
+            const currentUrl = new URL(this.props.linkElement.href);
+            if (
+                window.location.hostname === currentUrl.hostname &&
```

Comment:
If you want to access to the main location, you should always use browser.location (It s patchable for the test)
If you wan to acces to the iframe location you can t use the main window. (each plugin has access to the editable window)